File: app/Models/Pelanggan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Pelanggan extends Model
{
    use HasFactory;

    protected $table = 'pelanggan';

    protected $fillable = [
        'user_id',
        'nomor_pelanggan',
        'nama',
        'alamat',
        'nomor_meter',
        'no_telepon',
        'golongan_tarif_id',
        'petugas_id',
        'foto_ktp',
        'koordinat',
        'is_active',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function petugas()
    {
        return $this->belongsTo(User::class, 'petugas_id');
    }

    public function golonganTarif()
    {
        return $this->belongsTo(GolonganTarif::class);
    }

    public function pemakaianAir()
    {
        return $this->hasMany(PemakaianAir::class);
    }

    public function tagihan()
    {
        return $this->hasMany(Tagihan::class);
    }

    public function pengaduan()
    {
        return $this->hasMany(Pengaduan::class);
    }
}

File: mobile-flutter/lib/models/pelanggan.dart
import 'golongan_tarif.dart';

class Pelanggan {
  final int id;
  final int userId;
  final String nomorPelanggan;
  final String nama;
  final String alamat;
  final String nomorMeter;
  final String? noTelepon;
  final int golonganTarifId;
  final String? fotoKtp;
  final String? koordinat;
  final bool isActive;
  final GolonganTarif? golonganTarif;

  Pelanggan({
    required this.id,
    required this.userId,
    required this.nomorPelanggan,
    required this.nama,
    required this.alamat,
    required this.nomorMeter,
    this.noTelepon,
    required this.golonganTarifId,
    this.fotoKtp,
    this.koordinat,
    required this.isActive,
    this.golonganTarif,
  });

  factory Pelanggan.fromJson(Map<String, dynamic> json) {
    return Pelanggan(
      id: json['id'],
      userId: json['user_id'],
      nomorPelanggan: json['nomor_pelanggan'] ?? '',
      nama: json['nama'] ?? '',
      alamat: json['alamat'] ?? '',
      nomorMeter: json['nomor_meter'] ?? '',
      noTelepon: json['no_telepon'],
      golonganTarifId: json['golongan_tarif_id'] ?? 0,
      fotoKtp: json['foto_ktp'],
      koordinat: json['koordinat'],
      isActive: json['is_active'] == true || json['is_active'] == 1,
      golonganTarif: json['golongan_tarif'] != null
          ? GolonganTarif.fromJson(json['golongan_tarif'])
          : null,
    );
  }
}
